astropy/nddata/tests/test_utils.py

```diff
@@ -316,3 +320,116 @@ def test_block_size_len(self):
+        assert c.slices_cutout == (slice(0, 2), slice(0, 2))
+
+    def test_cutout_partial_overlap(self):
+        shape = (3, 3)
+        c = Cutout2D(self.data, (0, 0), shape, mode='partial')
+        assert c.data.shape == (3, 3)
+        assert c.data[1, 1] == 0
+        assert c.slices_original == (slice(0, 2), slice(0, 2))
+        assert c.slices_cutout == (slice(1, 3), slice(1, 3))
+
+    def test_cutout_partial_overlap_fill_value(self):
+        shape = (3, 3)
+        fill_value = -99
+        c = Cutout2D(self.data, (0, 0), shape, mode='partial',
+                   fill_value=fill_value)
+        assert c.data.shape == (3, 3)
+        assert c.data[1, 1] == 0
+        assert c.data[0, 0] == fill_value
+
+    def test_invalid_mode(self):
+        with pytest.raises(ValueError):
+            Cutout2D(self.data, (0, 0), (3, 3), mode='invalid')
+
+    def test_copy(self):
+        data = np.copy(self.data)
+        c = Cutout2D(data, (2, 3), (3, 3))
+        xy = (0, 0)
+        value = 100.
+        c.data[xy] = value
+        xy_orig = c.to_original_position(xy)
+        yx = xy_orig[::-1]
+        assert data[yx] == value
+
+        data = np.copy(self.data)
+        c2 = Cutout2D(self.data, (2, 3), (3, 3), copy=True)
+        c2.data[xy] = value
+        assert data[yx] != value
+
+    def test_to_from_large(self):
+        position = (2, 2)
+        shape = (3, 3)
+        c = Cutout2D(self.data, position, shape)
+        xy = (0, 0)
+        result = c.to_cutout_position(c.to_original_position(xy))
+        assert_allclose(result, xy)
+
+    def test_skycoord_without_wcs(self):
+        with pytest.raises(ValueError):
+            Cutout2D(self.data, self.position, (3, 3))
+
+    def test_skycoord(self):
+        c = Cutout2D(self.data, self.position, (3, 3), wcs=self.wcs)
+        skycoord_original = self.position.from_pixel(c.center_original[1],
+                                                     c.center_original[0],
+                                                     self.wcs)
+        skycoord_cutout = self.position.from_pixel(c.center_cutout[1],
+                                                   c.center_cutout[0], c.wcs)
+        assert_allclose(skycoord_original.ra.value, skycoord_cutout.ra.value)
+        assert_allclose(skycoord_original.dec.value,
+                        skycoord_cutout.dec.value)
+
+    def test_skycoord_partial(self):
+        c = Cutout2D(self.data, self.position, (3, 3), wcs=self.wcs,
+                   mode='partial')
+        skycoord_original = self.position.from_pixel(c.center_original[1],
+                                                     c.center_original[0],
+                                                     self.wcs)
+        skycoord_cutout = self.position.from_pixel(c.center_cutout[1],
+                                                   c.center_cutout[0], c.wcs)
+        assert_allclose(skycoord_original.ra.value, skycoord_cutout.ra.value)
```

Comment:
Would it make sense to use [astropy.tests.helper.assert_quantity_allclose](http://astropy.readthedocs.org/en/latest/api/astropy.tests.helper.assert_quantity_allclose.html) for all these angle (i.e. quantity sub-class) asserts?